ios/RNMobileDeviceManager.m

```diff
@@ -73,4 +101,68 @@ - (NSDictionary *)constantsToExport
     }
 }
 
+
+RCT_EXPORT_METHOD(isAutonomousSingleAppModeSupported: (RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [MobileDeviceManager isASAMSupported:^(BOOL reply){
+        if (reply) {
+            resolve(@YES);
+        } else {
+            resolve(@NO);
+        }
+    }];
+
+}
+
+RCT_EXPORT_METHOD(isSingleAppModeEnabled: (RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
+{
+    if (UIAccessibilityIsGuidedAccessEnabled()) {
+        resolve(@YES);
```

Comment:
Should be able to `resolve(UIAccessibilityIsGuidedAccessEnabled())` (might need to wrap it with an `@`, not sure tho)